packages/flutter_tools/lib/src/commands/daemon.dart

```diff
@@ -505,7 +505,7 @@ class AppDomain extends Domain {
   }
 }
 
-typedef void _DeviceEventHandler(Device device);
+typedef Future<Null> _DeviceEventHandler(Device device);
```

Comment:
In the end, we'll be spawning OS processes (`adb`) that could return out of order.  So if the plugin cares about receiving these events in order, you should restore the code (or something that accomplishes the same thing)